011-kafka-canary.md

```diff
@@ -0,0 +1,40 @@
+# Strmixi Canary
+
+Implement a Kafka canary tool which will act as an indicator of whether Kafka clusters are operating correctly. This would be achieved by creating a canary topic and continously producing and consuming events on the topics. 
+
+## Current situation
+
+Currently  Strimzi does not have any canary style feature. It does have readiness checks for Kafka cluster which can be replaced/augmented with the canary.
+
+## Motivation
+
+The canary would provide the following:
+  
+  * Improved readiness check for Kafka clusters
+  * SLI(s) for service admins
+  * Metrics:
+    * End-to-end latency
+    * Succesfull/failed events
+    * Disk usage
+    * Disk errors
+
+## Proposal
+
+The canary would be implemented as part of the Kafka Roller component. 
+A single _canary_ topic would be created on creation of a Kafka cluster. The topic would have one partition for Kafka broker. Events would then be continously produced and consumed to/from the canary topic. The internal TLS 9091 port would be used to connect to the brokers.
+
+As this happens the relevant metrics would be collected and exported. Errors such as the inability to produce to the topic can then be used by the KafkaRoller to trigger a restart, acting as healthc check for the Kafka cluster. 
```

Comment:
I wonder if this paragraph should be a stretch/future goal & initially deliver the proposal as a passive & optional feature ?

Reply:
Definitely agree with this - seems like if the Canary proposal were to be accepted, it would be standalone first, and if appropriate then a followup proposal would suggest the use of it for the KafkaRoller

Reply:
I'm happy for this functionality to be a stretch goal. However, I believe the main reason for building the canary into the Strimzi (instead of an external one) was to use it in the Kafka Roller. @ppatierno, @tombentley How to you feel about this being a stretch goal?